src/bin/pgcopydb/copydb.c

```diff
@@ -633,12 +633,12 @@ copydb_init_specs(CopyDataSpec *specs,
 		.cfPaths = specs->cfPaths,
 		.pgPaths = specs->pgPaths,
 
-		.source_pguri = { 0 },
-		.target_pguri = { 0 },
+		.source_pguri = !IS_EMPTY_STRING_BUFFER(options->source_pguri) ? options->source_pguri : NULL,
+		.target_pguri = !IS_EMPTY_STRING_BUFFER(options->target_pguri) ? options->target_pguri : NULL,
```

Comment:
It seems to me that `options->source_pguri` (and target) are now dynamically allocated strings (via calls to `strdup`) and not string buffers anymore. The IS_EMPTY_STRING_BUFFER call is not necessary here, please just use:

```c
    .source_pguri = options->source_pguri,
    .target_pguri = options->target_pguri,
```